quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -829,6 +830,11 @@ trait Parsing extends ValueComputation {
 
   implicit class InsertReturnCapabilityExtension(capability: ReturningCapability) {
     def verifyAst(returnBody: Ast) = capability match {
+      case OutputClauseSupported => returnBody match {
+        case Filter(_, _, _) =>
```

Comment:
Giving it a new read, I noticed this should be `Query`, otherwise that would compile: 

```
quote { qr4.insert(lift(TestEntity4(1L))).returning(r => query[TestEntity4]) }
```